File: Day30/longest-subarray-with-maximum-bitwise-and.js
// Time: O(N)  ::  Space: O(1)
var longestSubarray = function(nums) {
    let maxSubArrayVal = 0, longestSubArrayLen = 0, res = 0;

    for (let num of nums) {
        if (maxSubArrayVal < num) {
            maxSubArrayVal = num;
            res = longestSubArrayLen = 0;
        }

        if (maxSubArrayVal === num) {
            longestSubArrayLen++;
        } else {
            longestSubArrayLen = 0;
        }

        res = Math.max(res, longestSubArrayLen);
    }

    return res;
};

File: Day30/longest-subarray-with-maximum-bitwise-and.py
# Time: O(N)  ::  Space: O(1)

class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        maxSubArrayVal = longestSubArrayLen = res = 0
        for num in nums:
            if maxSubArrayVal < num:
                maxSubArrayVal = num
                res = longestSubArrayLen = 0

            if maxSubArrayVal == num:
                longestSubArrayLen += 1
            else:
                longestSubArrayLen = 0

            res = max(res, longestSubArrayLen)
        return res
        